mozilla/contain-facebook
# Fix #708 - Open PayPal inside the Facebook when donating

## Summary

This patch adds logic to open up a PayPal window inside the Facebook Container if the origin URL is `https://facebook.com/donate` and allows PayPal resources to be loaded inside Facebook Container. 

Privacy Considerations: 
- Does this patch leak info for users who have both PayPal and Facebook accounts but do not connect services? 
- Current allowed resources from PayPal include a wild card for subdomains. Is that okay? 

## Testing Steps

- Log into Facebook
  - _Note - Make sure you do not have PayPal already connected as a funding source_
- Go to https://www.facebook.com/fundraisers/ and select any active campaign
- Click "Donate" button 
- Under the Payment Method section, select "_Connect to PayPal_"
- Click blue  "Donate" button at the bottom of the form
- EXPECTED: A separate PayPal window should open and be **INSIDE** the Facebook Container
